service/history/shard/context_impl.go

```diff
@@ -1326,7 +1377,11 @@ func (s *ContextImpl) handleWriteErrorAndUpdateMaxReadLevelLocked(err error, new
 	case *persistence.CurrentWorkflowConditionFailedError,
 		*persistence.WorkflowConditionFailedError,
 		*persistence.ConditionFailedError,
-		*serviceerror.ResourceExhausted:
+		*persistence.InvalidPersistenceRequestError,
+		*persistence.TransactionSizeLimitError,
+		*serviceerror.ResourceExhausted,
+		*serviceerror.NotFound,
+		*serviceerror.NamespaceNotFound:
```

Comment:
could this error return?

Reply:
I won't want to assume how persistence layer is implemented, maybe it will return this error in the future. But contract won't change: the write is not committed.